src/main/java/org/springframework/integration/kafka/core/Connection.java

```diff
@@ -0,0 +1,305 @@
+import kafka.cluster.Broker;
+import kafka.common.ErrorMapping;
+import kafka.common.OffsetMetadataAndError;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.OffsetCommitRequest;
+import kafka.javaapi.OffsetCommitResponse;
+import kafka.javaapi.OffsetFetchRequest;
+import kafka.javaapi.OffsetFetchResponse;
+import kafka.javaapi.OffsetRequest;
+import kafka.javaapi.OffsetResponse;
+import kafka.javaapi.PartitionMetadata;
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.TopicMetadataResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.MessageAndOffset;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.util.Assert;
+
+/**
+ * An connection to a Kafka broker.
+ *
+ * @author Marius Bogoevici
+ */
+public class Connection {
+
+	public static final String DEFAULT_CLIENT_ID = "spring.kafka";
+
+	public static final int DEFAULT_BUFFER_SIZE = 64 * 1024;
+
+	public static final int DEFAULT_SOCKET_TIMEOUT = 10 * 1000;
+
+	private static Log log = LogFactory.getLog(Connection.class);
+
+	private final AtomicInteger correlationIdCounter = new AtomicInteger(new Random(new Date().getTime()).nextInt());
+
+	private final SimpleConsumer simpleConsumer;
+
+	private BrokerAddress brokerAddress;
+
+	public Connection(BrokerAddress brokerAddress) {
+		this(brokerAddress, DEFAULT_CLIENT_ID);
+	}
+
+	public Connection(BrokerAddress brokerAddress, String clientId) {
+		this(brokerAddress, clientId, DEFAULT_BUFFER_SIZE, DEFAULT_SOCKET_TIMEOUT);
+	}
+
+	public Connection(BrokerAddress brokerAddress, String clientId, int bufferSize, int soTimeout) {
+		this.brokerAddress = brokerAddress;
+		this.simpleConsumer = new SimpleConsumer(brokerAddress.getHost(), brokerAddress.getPort(), soTimeout, bufferSize, clientId);
+	}
+
+	/**
+	 * The broker address for this consumer
+	 *
+	 * @return broker address
+	 */
+	public BrokerAddress getBrokerAddress() {
+		return brokerAddress;
+	}
+
+	public void close() {
+		this.simpleConsumer.close();
+	}
+
+	/**
+	 * Fetches data from Kafka.
+	 *
+	 * @return a combination of messages and errors, depending on whether the invocation was successful or not
+	 * @throws ConsumerException
+	 */
+	public Result<KafkaMessageBatch> fetch(FetchRequest... requests) throws ConsumerException {
+		FetchRequestBuilder fetchRequestBuilder = new FetchRequestBuilder();
+		for (FetchRequest request : requests) {
+			Partition partition = request.getPartition();
+			long offset = request.getOffset();
+			int maxSize = request.getMaxSizeInBytes();
+			fetchRequestBuilder.addFetch(partition.getTopic(), partition.getId(), offset, maxSize);
+		}
+		FetchResponse fetchResponse;
+		try {
+			fetchResponse = this.simpleConsumer.fetch(fetchRequestBuilder.build());
+		}
+		catch (Throwable t) {
```

Comment:
Actually, we need to catch Exception, because checked exceptions can be still thrown by this call, but they're removed from the `SimpleConsumer` method signatures in the Scala code.

I agree on Throwable being excessive here. My bad.